pkg/identity/store.go

```diff
@@ -62,6 +62,10 @@ func (s *httpIDStore) RegisterNode(node pkg.Identifier, farm pkg.FarmID, version
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode == http.StatusNotFound {
```

Comment:
I think one of the issues is that bcdb returns 500 if farmer id is not found, which is also need to be fixed in the mock to return proper error code

Reply:
indeed, and this is also part of this PR: https://github.com/threefoldtech/zos/pull/622/files#diff-cd866ca7f60c9a054ceee3a4b934b6b2R38